cmd/ike/cmd/develop.go

```diff
@@ -2,53 +2,106 @@ package cmd
 
 import (
 	"fmt"
+	"os"
 	"os/exec"
+	"path/filepath"
 	"strings"
 
 	"github.com/aslakknutsen/istio-workspace/cmd/ike/config"
+	"github.com/aslakknutsen/istio-workspace/cmd/ike/watch"
 
+	"github.com/fsnotify/fsnotify"
 	gocmd "github.com/go-cmd/cmd"
 	"github.com/spf13/cobra"
 )
 
 const telepresenceBin = "telepresence"
 
+var streamOutput = gocmd.Options{
+	Buffered:  false,
+	Streaming: true,
+}
+
 func NewDevelopCmd() *cobra.Command {
 
 	developCmd := &cobra.Command{
 		Use:   "develop",
-		Short: "starts the development flow",
+		Short: "Starts the development flow",
 
 		PreRunE: func(cmd *cobra.Command, args []string) error { //nolint[:unparam]
 			if !telepresenceExists() {
 				return fmt.Errorf("unable to find %s on your $PATH", telepresenceBin)
 			}
-			config.SyncFlag(cmd, "deployment")
-			config.SyncFlag(cmd, "run")
-			config.SyncFlag(cmd, "port")
-			config.SyncFlag(cmd, "method")
 
-			return nil
+			return config.SyncFlags(cmd)
 		},
 		RunE: func(cmd *cobra.Command, args []string) error { //nolint[:unparam]
-			options := gocmd.Options{
-				Buffered:  false,
-				Streaming: true,
-			}
-			tp := gocmd.NewCmdOptions(options, telepresenceBin, parseArguments(cmd)...)
 
-			go redirectStreamsToCmd(tp, cmd)
+			if err := build(cmd); err != nil {
+				return err
+			}
 
-			tpStatusChan := tp.Start()
-			finalStatus := <-tpStatusChan
+			done := make(chan gocmd.Status)
+			runTp := make(chan struct{})
+
+			if cmd.Flag("watch").Changed {
+				slice, e := cmd.Flags().GetStringSlice("watch")
+				if e != nil {
+					return e
+				}
+
+				w, err := watch.NewWatch().
+					WithHandler(func(event fsnotify.Event, done chan<- struct{}) error {
+						_, _ = cmd.OutOrStdout().Write([]byte(event.Name + " changed. Restarting process.\n"))
```

Comment:
This is temporary workaround to be able to inspect the output in the buffer used in tests.

I will work on the proper solution next.